src/parsing.jl

```diff
@@ -393,25 +419,26 @@ end
 
 ## arrays
 # numeric arrays never have double quotes and always use ',' as a separator
-parse_numeric_element(::Type{T}, str) where T = parse(T, str)
+parse_numeric_element(::Type{T}, str) where {T} = parse(T, str)
 
-parse_numeric_element(::Type{Union{T, Missing}}, str) where T =
-    str == "NULL" ? missing : parse(T, str)
+function parse_numeric_element(::Type{Union{T,Missing}}, str) where {T}
+    return str == "NULL" ? missing : parse(T, str)
+end
 
-function parse_numeric_array(eltype::Type{T}, str::AbstractString) where T
+function parse_numeric_array(eltype::Type{T}, str::AbstractString) where {T}
```

Comment:
Don't put where clauses in brackets if they are in long-form function definitions and fit on a single line.